src/history/historyContainer.ts

```diff
@@ -0,0 +1,42 @@
+import { connect } from 'react-redux'
+import { ThunkDispatch } from 'redux-thunk'
+import { API, GlobalState } from 'types'
+
+import History from './history'
+import {
+  ChangeDateFilterAction,
+  ChangePriceFilterAction,
+  getHistoryByDateFilter,
+  getHistoryByPriceFilter
+} from './historyActions'
+
+const mapStateToProps = (state: GlobalState) => ({
+  company: state.company.symbol,
+  history: state.history.history,
+  priceFilter: state.history.priceFilter,
+  dateFilter: state.history.dateFilter
+})
+
+type HistoryContainerStateProps = ReturnType<typeof mapStateToProps>
+
+const mapDispatchToProps = (
+  dispatch: ThunkDispatch<GlobalState, API, ChangeDateFilterAction | ChangePriceFilterAction>
```

Comment:
You can perhaps create a new type that combine the union type you are doing here. Might make it easier to see.